src/audio/audio.c

```diff
@@ -164,49 +186,53 @@ int sound_play_panned(int sound_id, int ebx, sint16 x, sint16 y, sint16 z)
 *
 *  rct2: 0x006BD0F8
 */
-void start_title_music()
+void audio_start_title_music()
 {
-	int musicPathId;
-	switch (gConfigSound.title_music) {
-	default:
+	if (gGameSoundsOff || !RCT2_GLOBAL(RCT2_ADDRESS_SCREEN_FLAGS, uint8) & SCREEN_FLAGS_TITLE_DEMO) {
```

Comment:
You are missing the brackets on the second statement so I think the '!' will be on the screen flags instead of the the result.